test/tests/p5.SoundRecorder.js

```diff
@@ -53,18 +53,21 @@ describe('p5.SoundRecorder', function () {
 
     const mic = new p5.AudioIn();
     mic.start(function () {
+      recorder.setInput(mic);
       const outputSoundFile = new p5.SoundFile();
-      recorder.record(outputSoundFile, recordingDuration, function () {
-        expect(outputSoundFile.duration()).to.eq(recordingDuration);
-
-        const outputChannel = outputSoundFile.buffer.getChannelData(0);
-        expect(outputChannel[0]).to.not.eq(0);
-
-        outputSoundFile.dispose();
-        mic.dispose();
-        p5.prototype.outputVolume(0);
-        done();
-      });
+      setTimeout(() => {
+        recorder.record(outputSoundFile, recordingDuration, function () {
+          expect(outputSoundFile.duration()).to.eq(recordingDuration);
+
+          const outputChannel = outputSoundFile.buffer.getChannelData(0);
+          expect(outputChannel[0]).to.not.eq(0);
+
+          outputSoundFile.dispose();
+          mic.dispose();
+          p5.prototype.outputVolume(0);
+          done();
+        });
+      }, 100);
```

Comment:
why do we need this setTimeout?

Reply:
This time, the mic is taking some time to connect to the recorder in [this](https://github.com/processing/p5.js-sound/pull/626/files/1b60962f84a3eb68223edf7e74c446cce40399ed#diff-14453cfad7e58ed0e24d64543f6ff750b682d7965d960ff7c8acf40130f19d46R56) line(line 56). If we don't use the timeout, the `outputChannel` is returning all 0's, failing the test.

Reply:
got it. It would sure be useful if that method returned a Promise or had a callback